code/modules/vehicles/hardpoints/primary/arc_sentry.dm

```diff
@@ -0,0 +1,317 @@
+	ammo.current_rounds--
+	SEND_SIGNAL(new_bullet, COMSIG_BULLET_USER_EFFECTS, user)
+	new_bullet.runtime_iff_group = faction_group // Technically shouldn't be directly modifying this, but sue me
+
+	// turf-targeted projectiles are fired without scatter, because proc would raytrace them further away
+	var/ammo_flags = new_bullet.ammo.flags_ammo_behavior | new_bullet.projectile_override_flags
+	if(!HAS_FLAG(ammo_flags, AMMO_HITS_TARGET_TURF) && !HAS_FLAG(ammo_flags, AMMO_EXPLOSIVE)) //AMMO_EXPLOSIVE is also a turf-targeted projectile
+		new_bullet.scatter = scatter
+		target = simulate_scatter(new_bullet, target, origin_turf, get_turf(target), user)
+
+	INVOKE_ASYNC(new_bullet, TYPE_PROC_REF(/obj/projectile, fire_at), target, user, src, new_bullet.ammo.max_range, new_bullet.ammo.shell_speed)
+	new_bullet = null
+
+	shots_fired++
+	play_firing_sounds()
+	if(use_muzzle_flash)
+		muzzle_flash(Get_Angle(origin_turf, target))
+
+	set_fire_cooldown(gun_firemode)
+	return AUTOFIRE_CONTINUE
+
+
+/obj/item/hardpoint/primary/arc_sentry/generate_bullet(turf/origin_turf)
+	var/obj/projectile/arc_sentry/new_proj = new(origin_turf, create_cause_data(initial(name), owner))
+	new_proj.generate_bullet(new ammo.default_ammo)
+	new_proj.permutated += owner
+	// Apply bullet traits from gun
+	for(var/entry in traits_to_give)
+		var/list/trait_list
+		// Check if this is an ID'd bullet trait
+		if(istext(entry))
+			trait_list = traits_to_give[entry].Copy()
+		else
+			// Prepend the bullet trait to the list
+			trait_list = list(entry) + traits_to_give[entry]
+		new_proj.apply_bullet_trait(trait_list)
+	return new_proj
+
+/obj/item/hardpoint/primary/arc_sentry/start_fire(datum/source, atom/object, turf/location, control, params)
+	if(istype(object, /atom/movable/screen))
+		return
+
+	if(QDELETED(object))
+		return
+
+	if(!auto_firing && !burst_firing && !COOLDOWN_FINISHED(src, fire_cooldown))
+		if(max(fire_delay, burst_delay + extra_delay) >= 2.0 SECONDS) //filter out guns with high firerate to prevent message spam.
+			to_chat(source, SPAN_WARNING("You need to wait [SPAN_HELPFUL(COOLDOWN_SECONDSLEFT(src, fire_cooldown))] seconds before [name] can be used again."))
+		return
+
+	set_target(object)
+
+	if(gun_firemode == GUN_FIREMODE_SEMIAUTO)
+		var/fire_return = try_fire(object, source, params)
+		//end-of-fire, show ammo (if changed)
+		if(fire_return == AUTOFIRE_CONTINUE)
+			reset_fire()
+			display_ammo(source)
+	else
+		SEND_SIGNAL(src, COMSIG_GUN_FIRE)
+
+/obj/item/hardpoint/primary/arc_sentry/proc/get_target(atom/movable/new_target)
+	if(!islist(targets))
+		return
+	if(!targets.Find(new_target))
+		targets.Add(new_target)
+
+	if(!length(targets))
+		return
+
+	var/list/conscious_targets = list()
+	var/list/unconscious_targets = list()
+
+	for(var/atom/movable/movable as anything in targets) // orange allows sentry to fire through gas and darkness
+		if(isliving(movable))
+			var/mob/living/living_mob = movable
```

Comment:
In theory, `targets` can only contain `/mob/living`, since the only thing adding to the list is this:
https://github.com/cmss13-devs/cmss13/blob/9be48076449e904c9cd150db63fe319459d4f113/code/modules/vehicles/hardpoints/primary/arc_sentry.dm#L73-L75
It should be safe to change this loop
```dm
	for(var/mob/living/living_target as anything in targets)
		if(living_target.stat == DEAD)
etc etc
```